core/src/main/java/jeeves/xlink/Processor.java

```diff
@@ -487,5 +487,9 @@ private static void cleanXLinkAttributes(Element element, String action) {
         }
     }
 
+    public static void setLocalServiceInvoker(SpringLocalServiceInvoker springLocalServiceInvoker) {
+        localServiceInvoker = springLocalServiceInvoker;
```

Comment:
So I am trying to follow this logic:

1. registerToProcessor() is used to set "this" as the single static instance post constructor
2. This is not thread local, is that going to be a problem? Probably not since we spring context is being managed here
3. Do we need any kind of listener to clean up this static reference when spring is shutting down (to prevent leak if service is restarting or used in tests)

Reply:
Reviewing this change for use of `srvContext` being removed, looks like it used for `srvContext.getBean(String)` lookup...

Reply:
> So I am trying to follow this logic:
> 
>     1. registerToProcessor() is used to set "this" as the single static instance post constructor
> 
>     2. This is not thread local, is that going to be a problem? Probably not since we spring context is being managed here
> 
>     3. Do we need any kind of listener to clean up this static reference when spring is shutting down (to prevent leak if service is restarting or used in tests)

Most of Processor methods are static, and I don't know how to autowire simply in a very static context... (I Think there is no need for a listener, spring local service invoker (kind of a bad workaround for performances issues) is really bound to spring context and there should be one and only one, reentrant)